bundles/org.openhab.binding.comfoair/src/main/java/org/openhab/binding/comfoair/internal/ComfoAirSerialConnector.java

```diff
@@ -0,0 +1,586 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Connector class for serial communication with ComfoAir device
+ *
+ * @author Hans Böhm - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class ComfoAirSerialConnector {
+
+    private final Logger logger = LoggerFactory.getLogger(ComfoAirSerialConnector.class);
+
+    private static byte CTRL = (byte) 0x07;
+    private static byte[] START = { CTRL, (byte) 0xf0 };
+    private static byte[] END = { CTRL, (byte) 0x0f };
+    private static byte[] ACK = { CTRL, (byte) 0xf3 };
+
+    private static int RS232_ENABLED_VALUE = 0x03;
+    private static int RS232_DISABLED_VALUE = 0x00;
+
+    private boolean isSuspended = true;
+
+    private boolean connected = false;
+    private final String serialPortName;
+    private final int baudRate;
+    private final SerialPortManager serialPortManager;
+    private @Nullable SerialPort serialPort;
+    private @Nullable InputStream inputStream;
+    private @Nullable OutputStream outputStream;
+
+    public ComfoAirSerialConnector(final SerialPortManager serialPortManager, final String serialPortName,
+            final int baudRate) {
+        this.serialPortManager = serialPortManager;
+        this.serialPortName = serialPortName;
+        this.baudRate = baudRate;
+    }
+
+    /**
+     * Open serial port.
+     *
+     * @throws PortInUseException, UnsupportedCommOperationException, IOException
+     */
+    public void open() throws PortInUseException, UnsupportedCommOperationException, IOException {
+        logger.debug("open(): Opening ComfoAir connection");
+
+        try {
+            SerialPortIdentifier portIdentifier = serialPortManager.getIdentifier(serialPortName);
+            if (portIdentifier != null) {
+                SerialPort serialPort = portIdentifier.open(this.getClass().getName(), 3000);
+                serialPort.setSerialPortParams(baudRate, SerialPort.DATABITS_8, SerialPort.STOPBITS_1,
+                        SerialPort.PARITY_NONE);
+                serialPort.enableReceiveThreshold(1);
+                serialPort.enableReceiveTimeout(1000);
+                serialPort.notifyOnDataAvailable(true);
+                this.serialPort = serialPort;
+
+                inputStream = new DataInputStream(new BufferedInputStream(serialPort.getInputStream()));
+                outputStream = serialPort.getOutputStream();
+
+                ComfoAirCommand command = ComfoAirCommandType.getChangeCommand(ComfoAirCommandType.ACTIVATE.getKey(),
+                        OnOffType.ON);
+
+                if (command != null) {
+                    sendCommand(command, ComfoAirCommandType.Constants.EMPTY_INT_ARRAY);
+                } else {
+                    logger.debug("Failure while creating COMMAND: {}", command);
+                }
+                setConnected(true);
+            } else {
+                logger.debug("open(): No such Port: {}", serialPortName);
```

Comment:
Would be good to throw an exception here, too, to let the user know what's wrong via the UI.

Consider whether the connected flag is still needed when using exceptions for error handling. If I see correctly, the flag is always false if you throw an exception, which would be redundant.

Reply:
done